src/sync/remote_mongo_collection.cpp

```diff
@@ -21,461 +21,411 @@
 {
-    count(filter_json, 0, completion_block);
+    count(filter_bson, 0, completion_block);
 }
 
-void RemoteMongoCollection::insert_many(std::vector<std::string> documents,
-                                        std::function<void(std::vector<std::string>, util::Optional<AppError>)> completion_block)
+void RemoteMongoCollection::insert_many(bson::BsonArray documents,
+                                        std::function<void(std::vector<ObjectId>,
+                                                           util::Optional<AppError>)> completion_block)
 {
-     try {
-
-         auto documents_array = nlohmann::json::array();
-         for (std::string& document_json : documents) {
-             documents_array.push_back(nlohmann::json::parse(document_json));
-         }
-
-         auto base_args = m_base_operation_args;
-         base_args.push_back({"documents", documents_array});
-         auto args = nlohmann::json({{"arguments", nlohmann::json::array({base_args})}});
-
-         m_service.call_function("insertMany",
-                                  args.dump(),
-                                  [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
-             if (value && !error) {
-                 try {
-                     auto json = nlohmann::json::parse(*value);
-                     auto inserted_ids = json.at("insertedIds").get<std::vector<nlohmann::json>>();
-                     auto oid_array = std::vector<std::string>();
-                     for(auto& inserted_id : inserted_ids) {
-                         auto oid = inserted_id.at("$oid").get<std::string>();
-                         oid_array.push_back(oid);
-                     }
-                     return completion_block(oid_array, error);
-                 } catch (const std::exception& e) {
-                     return completion_block(std::vector<std::string>(), AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
-                 }
-             }
-
-             return completion_block(std::vector<std::string>(), error);
-         });
-    } catch (const std::exception& e) {
-        return completion_block(std::vector<std::string>(), AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
-    }
+    auto base_args = bson::BsonDocument(m_base_operation_args);
+    base_args["documents"] = documents;
+    
+    m_service->call_function("insertMany",
+                             bson::BsonArray({base_args}),
+                             m_service_name,
+                             [completion_block](util::Optional<AppError> error, util::Optional<bson::Bson> value) {
+        if (error) {
+            return completion_block({}, error);
+        }
+
+        auto bson = static_cast<bson::BsonDocument>(*value);
+        auto inserted_ids = static_cast<bson::BsonArray>(bson["insertedIds"]);
+        auto oid_array = std::vector<ObjectId>();
```

Comment:
`std::vector<ObjectId>(inserted_ids.begin(), inserted_ids.end())` is probably sufficient to populate this vector.